WIP Functional Test Thread

BenWhitehead opened the pull request.

Commit:
Create skeleton for functional test server

ConnorDoyle commented on functional-tests/server/project/Build.scala:
```diff
@@ -0,0 +1,113 @@
+import scalariform.formatter.preferences._
+import sbtunidoc.Plugin._
+
+object ServiceNetFunctionalTestsServerBuild extends Build {
```
Let's make this whole thing a subproject in the main `project/Build.scala`.  This will make it easier to run the integration tests (because it won't be necessary to first publish the `service-net` artifacts).

Commit:
Update build config based on feedback from connor

Commit:
Create skeleton testing client and very basic test.

Configurable with properties
* test.client.connect.hostname
* test.client.connect.port

Commit:
Update server to use properties for addr and port binding
* test.server.connect.addr
* test.server.connect.port

Commit:
Formatting and improved log message for simple test

Commit:
Update assembly to contain functional tests

Commit:
Simplify property names for Server config

Commit:
Refactor test out of Client and into BalanceFactorTest.scala which now reports with an object rather than just logging information.

Update client to expect an application parameter for the hostname[:port] to connect to.
Cleanup formatting for column width.

BenWhitehead commented:
_Currently, both the test server and test client are bundled into the app's uber-jar._

##### To run the test server run the following command in the project root:

``` bash
java -classpath target/scala-2.10/service-net-assembly-0.1.0-SNAPSHOT.jar -Dhttp.ip=::1 -Dhttp.port=9797 mesosphere.servicenet.tests.Server
```

##### To run the client, run the following command in the project root

``` bash
java -classpath target/scala-2.10/service-net-assembly-0.1.0-SNAPSHOT.jar mesosphere.servicenet.tests.Client ipv6-localhost:9797
```

###### Additional config properties for the client

```
test.client.balanceTest.request.count[10000]
test.client.balanceTest.expect.balanceFactor[100.0]
test.client.balanceTest.expect.balanceFactor.delta[0.1]
```

Commit:
Create test-tools script and scala wrapper that will be used to invoke specific commands needed for the test.

Commit:
Create a util to template the creation of the Doc needed to test with 2 machines.

solidsnack committed:
Formatting, import ordering.

solidsnack committed:
Formatting, property names.

Commit:
Slightly rework BalanceFactorTest.scala to derive what the balance factor should be based on the number of servers it received responses from. Change client params to removed explicit balance factor.

Add logging to report client configuration values when it's ran.

Update and cleanup format of the json object that is reported to stdout.

Ensure everything is reported in factors now rather than a mixture of factor and percentage.

Commit:
Rename project to tests from functional-tests.

Create Wrapper main that can dispatch to the client or server.

Commit:
Update server to serve a 10mb byte array for each request to facilitate heavy bandwidth usage.
